src/librustc/hir/mod.rs

```diff
@@ -1983,9 +1984,9 @@ pub struct InlineAsmOutput {
 pub struct InlineAsm {
     pub asm: Symbol,
     pub asm_str_style: StrStyle,
-    pub outputs: HirVec<InlineAsmOutput>,
-    pub inputs: HirVec<Symbol>,
-    pub clobbers: HirVec<Symbol>,
+    pub outputs: Vec<InlineAsmOutput>,
+    pub inputs: Vec<Symbol>,
+    pub clobbers: Vec<Symbol>,
```

Comment:
Unclear why these are vectors.
Are they mutated in HIR somehow?

Reply:
It's to keep `Clone` on `InlineAsm`, I believe `mir::StatementKind::InlineAsm` contains a `hir::InlineAsm`? So in a sense this `struct` is not specific to HIR.

Reply:
Could you add an explanation comment?